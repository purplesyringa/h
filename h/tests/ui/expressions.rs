fn main() {
    let invert = h::map! {
        !0_i32 => (),
    };

    let negate = h::map! {
        -"meow" => (),
    };

    let deref = h::map! {
        *0 => (),
    };

    let what = h::map! {
        if false { main() } => ()
    };
}
