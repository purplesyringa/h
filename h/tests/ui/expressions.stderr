error: unsupported unary operator
       `h` is not a full-blown interpreter; only certain operators are allowed
 --> tests/ui/expressions.rs:3:9
  |
3 |         !0_i32 => (),
  |         ^^^^^^

error: cannot apply unary operator `-` to type `&_`
 --> tests/ui/expressions.rs:7:9
  |
7 |         -"meow" => (),
  |         ^^^^^^^

error: type `{integer}` cannot be dereferenced
  --> tests/ui/expressions.rs:11:9
   |
11 |         *0 => (),
   |         ^^

error: unsupported expression
       `h` is not a full-blown interpreter; only certain operators are allowed
  --> tests/ui/expressions.rs:15:9
   |
15 |         if false { main() } => ()
   |         ^^^^^^^^^^^^^^^^^^^
